storage/scratchpad/src/sparse_merkle/mod.rs

```diff
@@ -97,10 +105,33 @@ pub enum AccountStatus<V> {
     Unknown,
 }
 
-/// The Sparse Merkle Tree implementation.
+/// The inner content of a sparse merkle tree, we have this so that even if a tree is dropped, the
+/// INNER of it can still live if referenced by a later version.
 #[derive(Debug)]
+struct Inner<V> {
+    /// Reference to the root node, initially a strong reference, and once pruned, becomes a weak
+    /// reference, allowing nodes created by this version to go away.
+    root: ArcSwap<SubTree<V>>,
+    /// Reference to the INNER base tree, needs to be a strong reference if the base is speculative
+    /// itself, so that nodes referenced in this version won't go away because the base tree is
+    /// dropped.
+    base: ArcSwapOption<Inner<V>>,
+}
+
+impl<V: CryptoHash> Inner<V> {
+    fn prune(&self) {
+        // Replace the link to the root node with a weak reference, so all nodes created by this
+        // version can be dropped.
+        self.root.store(Arc::new(self.root.load().as_ref().weak()));
```

Comment:
Explain why `root` is replaced with a weak reference, instead of being replaced by `None` like `base`?

Maybe you still want to read from the tree after it's pruned? Some sort of caching?